Add support for music rsync

wagner-ch opened the pull request:
Somehow the rsync_archive was missing the copy-music.sh script. Someone else already tried it, but you rejected his PR 
(see https://github.com/marcone/teslausb/pull/641)
I checked his script and also looked into your script from cifs_archive and found it quite funny, that you are using rsync in cifs archive anyway. So I took your script, changed SRC to use the available RSYNC variables from setup and just removed some cifs specific parts. 
I also adjusted the install script to be more generic on copying the copy-music.sh script.
From what I saw, I think this should be complete now, as everything else is already implemented quite generic.

Commit:
Created copy-music.sh for rsync archive.

Commit:
Be more generic on copying copy-music.sh script.

marcone commented on run/rsync_archive/copy-music.sh:
```diff
@@ -0,0 +1,80 @@
+  do
+    for _ in {1..10}
+    do
+      if timeout 3 /root/bin/archive-is-reachable.sh "$ARCHIVE_SERVER"
```
This should either use `RSYNC_SERVER`, or this entire function should be removed and a timeout added to the rsync call, similar to what archive-clips.sh does.

wagner-ch replied:
But `ARCHIVE_SERVER` is set to `RSYNC_SERVER` in script `archiveloop`, if in config `ARCHIVE_SYSTEM` is set to `rsync`. So this is "generic" and works as expected.

marcone replied:
ah yes, you are correct

marcone commented on setup/pi/configure.sh:
```diff
@@ -225,7 +225,7 @@ function install_archive_scripts () {
   copy_script "$archive_module"/disconnect-archive.sh "$install_path"
   copy_script "$archive_module"/archive-is-reachable.sh "$install_path"
-  if [ -n "${MUSIC_SHARE_NAME:+x}" ] && grep cifs <<< "$archive_module"
+  if [ -n "${MUSIC_SHARE_NAME:+x}" ] && [ -e "$archive_module"/copy-music.sh ]
```
The config file documents `MUSIC_SHARE_NAME` as being the name of the CIFS share that holds the music to copy. For rsync there there is no CIFS share, so either there should be a separate variable to indicate that music should be rsync'd, or the comment in the config file should be updated.

wagner-ch replied:
Well, as I for example set  `ARCHIVE_SYSTEM` to `rsync` as the comment in `teslausb_setup_variables.conf` states to do, that was the start of my confusion. I was expecting it to be the same and will adjust the comment then of course, as there is anway no way to use different archive systems for music and cam.

Commit:
Removed connection monitor and using rsync builtin timeout now.

- also log what we do to separate file
- renamed old 'LOG' file to 'STATS', as it is a stats file
- added -no-perms option to rsync
- added -L option to rsync

Commit:
Adjusted comment in config sample

marcone commented:
I've reverted this change because it was incomplete and didn't actually work:
- it didn't correctly copy the copy-music.sh script (I fixed this in [this change ](https://github.com/marcone/teslausb/commit/f4d9359f53088fe1f16a9d9a29a170a5ba68a02f), also reverted now)
- archiveloop checks whether music is syncing is configured by checking the existence of /mnt/musicarchive, but that folder only gets created for cifs, so it never ran the copy script for rsync.
Given that I clearly did a bad job reviewing, I've reverted this change, and will strive to be more careful  in my reviews from now on. Feel free to resubmit the change, but please test that it works on a fresh install.

wagner-ch commented:
Thx for reporting back. I'll check, which changes I must have done manually, as it clearly works on  my Installation. But indeed it hasn't been a clean/fresh install. I tried to reproduce all the manual changes with my commits. 

On Sunday, 23 April 2023, marcone wrote:
> I've reverted this change because it was incomplete and didn't actually work:
> - it didn't correctly copy the copy-music.sh script (I fixed this in [change ](https://github.com/marcone/teslausb/commit/f4d9359f53088fe1f16a9d9a29a170a5ba68a02f), also reverted now)
> - archiveloop checks whether music is syncing is configured by checking the existence of /mnt/musicarchive, but that folder only gets created for cifs, so it never ran the copy script for rsync.
> Given that I clearly did a bad job reviewing, I've reverted this change, and will strive to be more careful  in my reviews from now on. Feel free to resubmit the change, but please test that it works on a fresh install.
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/marcone/teslausb/pull/758#issuecomment-1519110131
> You are receiving this because you authored the thread.
> 
> Message ID: ***@***.***

-- 
Beste Grüße,

Christian Wagner

flosoft commented:
So, it seems like I got this running with the files in this commit.

The only additional change I made was in the following line: https://github.com/marcone/teslausb/blob/8de2a611d04e24a1af1655e44789d0d74f6cc5b3/run/archiveloop#L653

I just changed this to:   `if timeout 5 [ -d "$MUSIC_MOUNT" ]`

Once done, it all seems to work.
